src/foam/lib/json/Outputter.java

```diff
@@ -232,11 +244,90 @@ protected Boolean maybeOutputProperty(FObject fo, PropertyInfo prop, boolean inc
     if ( mode_ == OutputterMode.STORAGE && prop.getStorageTransient() ) return false;
     if ( ! outputDefaultValues_ && ! prop.isSet(fo) ) return false;
 
+    Object value = prop.get(fo);	
+    if ( value == null ) return false;
+
     if ( includeComma ) writer_.append(",");
     outputProperty(fo, prop);
     return true;
   }
 
+  protected void outputFObjectDelta(FObject o, FObject n, boolean outerObject) {
+    ClassInfo info = o.getClassInfo();
+    boolean outputComma = true;
+    if ( ! o.equals(n) ) {
```

Comment:
Transient fields could be the only fields that are different which would cause this to output.